src/crl.c

```diff
@@ -99,8 +106,24 @@ static int InitCRL_Entry(CRL_Entry* crle, DecodedCRL* dcrl, const byte* buff,
         return WOLFSSL_FAILURE;
     }
 #endif
-    crle->certs = dcrl->certs;   /* take ownsership */
+#ifdef CRL_STATIC_REVOKED_LIST
+    int totalCerts = dcrl->totalCerts;
+    RevokedCert* rc = dcrl->certs;
+
+    if (totalCerts > CRL_MAX_REVOKED_CERTS)
+        return WOLFSSL_FAILURE;
+
+    while (totalCerts > 0 && rc != NULL) {
```

Comment:
This will be very slow.
Can we not change GetRevoked()?

Reply:
@SparkiDev, Good point!! The initial focus was on improving the lookup side. Implemented the required changes to improve the caching/storing of data as well.